keras_cv/layers/preprocessing/aug_mix.py

```diff
@@ -0,0 +1,295 @@
+        alpha: a float value used as the probability coefficients for the
+            Beta and Dirichlet distributions. Defaults to 1.0.
+        seed: Integer. Used to create a random seed.
+
+    References:
+        [AugMix paper](https://arxiv.org/pdf/1912.02781)
+        [Official Code](https://github.com/google-research/augmix)
+        [Unoffial TF Code](https://github.com/szacho/augmix-tf)
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    augmix = keras_cv.layers.preprocessing.mix_up.AugMix([0, 255])
+    augmented_images = augmix(images)
+    ```
+    """
+
+    def __init__(
+        self,
+        value_range,
+        severity=3,
+        width=3,
+        depth=-1,
+        alpha=1.0,
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(seed=seed, **kwargs)
+        self.value_range = value_range
+        self.severity = severity
+        self.width = width
+        self.depth = depth
+        self.alpha = alpha
+        self.seed = seed
+        self.auto_vectorize = False
+
+    @staticmethod
+    def _sample_from_dirichlet(alpha):
+        gamma_sample = tf.random.gamma(shape=(), alpha=alpha)
+        return gamma_sample / tf.reduce_sum(gamma_sample, axis=-1, keepdims=True)
+
+    @staticmethod
+    def _sample_from_beta(alpha, beta):
+        sample_alpha = tf.random.gamma((), 1.0, beta=alpha)
+        sample_beta = tf.random.gamma((), 1.0, beta=beta)
+        return sample_alpha / (sample_alpha + sample_beta)
+
+    def _sample_level(self, level, maxval, dtype, shape=()):
+        level = self._random_generator.random_uniform(
+            shape=shape, minval=0.1, maxval=level, dtype=tf.float32
+        )
+        return tf.cast((level) * maxval / 10.0, dtype)
+
+    def _loop_on_depth(self, depth_level, image_aug):
+        op_index = self._random_generator.random_uniform(
+            shape=(), minval=0, maxval=8, dtype=tf.int32
+        )
+        image_aug = self._apply_op(image_aug, op_index)
+        depth_level += 1
+        return depth_level, image_aug
+
+    def _loop_on_width(self, image, chain_mixing_weights, curr_chain, result):
+        image_aug = tf.identity(image)
+        depth = tf.cond(
+            tf.greater(self.depth, 0),
+            lambda: self.depth,
+            lambda: self._random_generator.random_uniform(
+                shape=(), minval=1, maxval=3, dtype=tf.int32
+            ),
+        )
+
+        depth_level = tf.constant([0], dtype=tf.int32)
+        depth_level, image_aug = tf.while_loop(
+            lambda depth_level, image_aug: tf.less(depth_level, depth),
+            self._loop_on_depth,
+            [depth_level, image_aug],
+        )
+        result += tf.gather(chain_mixing_weights, curr_chain) * image_aug
+        curr_chain += 1
+        return image, chain_mixing_weights, curr_chain, result
+
+    def _auto_contrast(self, image):
+        return layers.AutoContrast(self.value_range)(image)
```

Comment:
We'll want to construct a single auto_contrast layer in the constructor instead.

If you construct a new one in the call method this can be problematic and lead to duplicate constructions.

prefer:
self.auto_contrast = layers.AutoContrast(value_range=self.value_range)